test/web.js

```diff
@@ -10,710 +10,801 @@ var testCreds = require("./testCredentials");
-    accessTokens: testCreds.accessTokens,
-    enableCSRF: true
-  });
+  lab.before(async () => {
+    s = await server({
+      loginAPI: "http://localhost:3232",
+      cookieSecret,
+      oauth_clients: testCreds.clients,
+      authCodes: testCreds.authCodes,
+      accessTokens: testCreds.accessTokens,
+      enableCSRF: false
+    });
+
+    s2 = await server({
+      loginAPI: "http://localhost:3232",
+      cookieSecret,
+      oauth_clients: testCreds.clients,
+      authCodes: testCreds.authCodes,
+      accessTokens: testCreds.accessTokens,
+      enableCSRF: true
+    });
 
-  var ls = loginServer({
-    loginAPI: "http://localhost:3232"
+    ls = loginServer({
+      loginAPI: "http://localhost:3232"
+    });
   });
 
-  lab.test("GET / Redirects to /signup", function(done) {
+  lab.test("GET / Redirects to /signup", async () => {
     var request = {
       method: "GET",
       url: "/"
     };
 
-    s.inject(request, function(response) {
-      Code.expect(response.statusCode).to.equal(302);
-      Code.expect(response.headers.location).to.equal("/signup");
-      done();
-    });
+    const response = await s.inject(request);
+    Code.expect(response.statusCode).to.equal(302);
+    Code.expect(response.headers.location).to.equal("/signup");
   });
 
-  lab.test("GET /signup returns 200, sets csrf cookie, and has security headers", function(done) {
+  lab.test("GET /signup returns 200, sets csrf cookie, and has security headers", async () => {
     var request = {
       method: "GET",
       url: "/signup"
     };
 
-    s2.inject(request, function(response) {
-      Code.expect(response.statusCode).to.equal(200);
-      Code.expect(response.headers["set-cookie"]).to.exist();
-      Code.expect(response.headers["set-cookie"]).to.match(/crumb=/);
+    const response = await s2.inject(request);
+    Code.expect(response.statusCode).to.equal(200);
+    Code.expect(response.headers["set-cookie"]).to.exist();
+    Code.expect(response.headers["set-cookie"]).to.match(/crumb=/);
+
+    Code.expect(response.headers["strict-transport-security"]).to.equal("max-age=15768000");
+    Code.expect(response.headers["x-xss-protection"]).to.equal("1; mode=block");
+    Code.expect(response.headers["x-frame-options"]).to.equal("DENY");
+    Code.expect(response.headers["x-download-options"]).to.equal("noopen");
+    Code.expect(response.headers["x-content-type-options"]).to.equal("nosniff");
+
+    Code.expect(response.headers["content-security-policy"]).to.equal(
+      "base-uri 'self';" +
+      "connect-src 'self';default-src 'none';font-src 'self' https://fonts.gstatic.com;" +
+      "img-src 'self' data: https://www.google-analytics.com http://www.google-analytics.com;" +
+      "script-src 'self' 'unsafe-eval' https://www.google-analytics.com http://www.google-analytics.com;" +
+      "style-src 'self' https://fonts.googleapis.com"
+    );
+  });
+
+  lab.test("POST Create User", async () => {
+    try {
```

Comment:
this pattern for launching the login server happens in all the tests that need it.